src/workloads/docs/CrudActorFSM.yml

```diff
@@ -0,0 +1,148 @@
+- Name: SetupDemo
+  Type: Crud
+  Threads: 1
+  Database: StateTest
+  Phases:
+  - Repeat: 1
+    Collection: CrudDemo
+    Operations:
+    - OperationName: bulkWrite
+      OperationCommand:
+        WriteOperations:
+        - WriteCommand: insertOne
+          Document: {phoneID: 0, state: "Off"}
+        - WriteCommand: insertOne
+          Document: {phoneID: 1, state: "Off"}
+        - WriteCommand: insertOne
+          Document: {phoneID: 2, state: "Off"}
+        - WriteCommand: insertOne
+          Document: {phoneID: 3, state: "Off"}
+        - WriteCommand: insertOne
+          Document: {phoneID: 4, state: "Off"}
+    - &nop {Nop: true}
+    - *nop
+
+- Name: StateSmartPhoneDemo
+  Type: Crud
+  Threads: 5
+  Database: StateTest
+  Phases:
+  - *nop
+  - Duration: 30 seconds
+    Collection: CrudDemo
+    States:
+    - Name: "On"  # The device is on
+
+      # The states support the keywords Repeat, SleepBefore, SleepAfter, (local) Rate, and
+      # Duration. These keywords behave the same as in Phase Control, but operate per state (i.e.,
+      # per actor instance as it goes through the state machine.
+      Repeat: 1
+      Operations:
+      # A state can have (optional) a set of operations that execute upon entering the state. This
+      # state updates a specific document to indicate it is in state On.
+
+      # Note: Following the analogy to phases -- phases may be blocking or non-blocking. States are
+      # always non-blocking. When the phase ends, the state machine will finish any open operations,
+      # and then exit. It if is waiting for a delay, it will exit immediately. This is required as
+      # there may be states with long delays.
+      # Question: Should we support having a single fire operation with repeating operations? Here I
+      # could invent a case where the following operation is repeat 1, but there is another
+      # operation that repeats once a second while in this state. I don't know how to represent
+      # that.  
+      - OperationName: updateOne
+        OperationCommand:
+          Filter: &filter {phoneID: {^ActorID: {}}}
+          Update:
+            $set: {state: "On"}
+
+      # Transitions specifies next states, delays, and operations. The On state can transition to
+      # each of the three other states
+      Transitions:
+      - To: "Off"  # Transition to this state
+
+        # Transitions are probabilistically chosen based on weight. A transition with weight 2 is
+        # twice as likely to be picked as a transition with weight 1. Weights must be greater than
+        # 0.
+        Weight: 1
+
+        # Transitions also support the phase keywords of sleepBefore, sleepAfter, and repeat.
+        # ^TimeSpec is a proposed generator to randomly generate acceptable TimeSpecs with units. It
+        # accepts any valid int or double generator or constant for value.
+        SleepBefore: {^TimeSpec: {value: {^RandomDouble: {min: 0, max: 15}}, unit: "seconds"}}
```

Comment:
On one hand, I feel like `SleepBefore` should be in the state not the transition. That would fit with adding other `PhaseLoop`-like syntax such as `Duration` or `Repeat` (which would only make sense in the state). On the other hand, `SleepBefore` in states wouldn't let you differentiate the time before transitioning for different transitions without adding intermediate states. Assuming we go the route of only having ops and such in the state.

Reply:
Exactly. That's hy I put the SleepBefore in the transition.